### PR TITLE
Feat/test and cleanup

### PR DESCRIPTION
**Pull request by oed**
* Refactored code into three modules: **RedisCache**, **CacheService**, and **Pinning**. 
* `src/node.js` is now the main bin file.
* Added tests for all modules
* Some tests missing for CacheService, but should be fine for now
* Added circleci config
* Should fix security vulnerabilities in this repo

### CODE REVIEW

**Comment by c-castillo** [comment]
@oed I had a `node-gyp` error when running it on my local setup. Just a fresh install of node 9

```
AttributeError: 'NoneType' object has no attribute 'groups'
gyp ERR! configure error 
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/Users/castelo/.nvm/versions/node/v9.11.2/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:336:16)
gyp ERR! stack     at ChildProcess.emit (events.js:180:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:209:12)
gyp ERR! System Darwin 18.2.0
gyp ERR! command "/Users/castelo/.nvm/versions/node/v9.11.2/bin/node" "/Users/castelo/.nvm/versions/node/v9.11.2/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/castelo/3box/3box-pinning-server/node_modules/tiny-secp256k1
gyp ERR! node -v v9.11.2
gyp ERR! node-gyp -v v3.6.2
gyp ERR! not ok 
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: gc-stats@1.2.1 (node_modules/gc-stats):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: gc-stats@1.2.1 install: `node-pre-gyp install --fallback-to-build`
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Exit status 1

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! tiny-secp256k1@1.0.1 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the tiny-secp256k1@1.0.1 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/castelo/.npm/_logs/2018-12-19T23_03_11_929Z-debug.log
```

Not sure if it's my end. I'll try it on a docker container to double check it's  (not) a problem with my setup.

**Comment by oed** [comment]
When do you get that error @c-castillo ? Looks like an install error?
I'm using node v10.11.0 but I don't think that should matter

**Comment by c-castillo** [comment]
With node 10 works without any problems. @oed

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+version: 2
+jobs:
+  build:
+    working_directory: ~/3box-pinning-server
+    docker:
+      - image: circleci/node:9
```
Comment (c-castillo) [inline .circleci/config.yml]: Reflect the correct version of node that is required

**Pull request merged**